### PR TITLE
Simplifications

### PR DESCRIPTION
**Pull request by Paul-Licameli**
Resolves: #6070
Resolves: #6072

Depends on
- #5965

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

QA
- [x] Wouldn't hurt to re-test every box of #5965 
- [x] Mix and render menu commands
- [x] Lengthening sync-lock adjustment of wave tracks
- [x] Loudness effect works correctly with mono tracks alone, and with stereo
- [x] Import Raw, with more than two channels, works as before
- [x] Change Speed effect
- [x] Effect preview play
- [x] Equalization
- [x] Generate silence
- [x] Noise Reduction
- [x] Paulstretch
- [x] Repeat effect
- [x] Change Pitch (high and low quality), Change Tempo (ditto), Sliding time/pitch shift effects
- [x] Compressor
- [x] At least one Nyquist effect and Nyquist generator
- [x] Cross-document copy/paste with hidden smart clip data
- [x] Edit > Duplicate
- [x] Split Cut
- [x] Split New
- [x] Edit > Labeled Audio menu commands
- [x] FIndClipping done once; then, done again on another track, re-using the same label track named "Clipping"
- [x] Import MIDI
- [x] Import Ffmpeg
- [x] Import FLAC
- [x] Import MP3
- [x] Import OGG
- [x] Import Opus
- [x] Import PCM
- [x] Import WavPack
- [x] Shift + Click in track control panel to extend the range of selected tracks (either upward or down)
- [x] Track focus changes (up and down arrow keys moving the yellow border)
- [x] Mix and render examples of stereo with nontrivial envelope
- [x] Cross-project copy and paste, where project tempi differ, do the correct stretch on insertion
- [x] Ctrl- and unmodified-clicks on track control panels change track selection correctly
- [x] Clicks of icon buttons in mixer board (with or without shift, or ctrl) too
- [x] Click of right channel of partly visible track at top of viewport, correctly scrolls it into view
- [x] Updates of vertical ruler width for changes of track height or waveform/spectrogram scale type behave as before
- [x] Mute and solo buttons work as before
- [x] Edit > Audio Clips > Split
- [x] Click in right channel to delete or expand CutLine

### CODE REVIEW

File: libraries/lib-wave-track/WaveTrack.cpp [hunk 1]
```diff
@@ -1133,6 +1133,13 @@ std::shared_ptr<WaveTrack> WaveTrackFactory::Create(sampleFormat format, double
...
 TrackListHolder WaveTrackFactory::Create(size_t nChannels)
+{
+   assert(nChannels > 0);
+   assert(nChannels <= 2);
+   return Create(nChannels, QualitySettings::SampleFormatChoice(), mRate.GetRate());
+}
+
+TrackListHolder WaveTrackFactory::CreateMany(size_t nChannels)
 {
    return Create(nChannels, QualitySettings::SampleFormatChoice(), mRate.GetRate());
```
Comment (saintmatthieu) [inline libraries/lib-wave-track/WaveTrack.cpp]: Shouldn't this be calling `CreateMany` ?

File: src/menus/TrackMenus.cpp [hunk 2]
```diff
@@ -74,7 +74,7 @@ void DoMixAndRender(AudacityProject &project, bool toNewTrack)
       }
 
       // Add new tracks
-      const bool stereo = newTracks->NChannels() > 1;
+      const bool stereo = (*newTracks->begin())->NChannels() > 1;
```
Comment (saintmatthieu) [inline src/menus/TrackMenus.cpp]: So now `newTracks.size() == 1` ?

File: libraries/lib-note-track/NoteTrack.cpp [hunk 3]
```diff
@@ -173,7 +173,7 @@ Alg_seq &NoteTrack::GetSeq() const
    return *mSeq;
 }
 
-TrackListHolder NoteTrack::Clone(bool) const
+Track::Holder NoteTrack::Clone(bool) const
```
Comment (saintmatthieu) [inline libraries/lib-note-track/NoteTrack.cpp:176]: Nice!

**Review by saintmatthieu — COMMENTED** [review]
Reviewed commits up to 43228cedb1ecbe62db4e99a4a973e059e342112b exclusive.
Best PR *ever* !!

**Comment by Paul-Licameli** [comment]
> Reviewed commits up to [43228ce](https://github.com/audacity/audacity/commit/43228cedb1ecbe62db4e99a4a973e059e342112b) exclusive. Best PR _ever_ !!

I have other candidates for that honor over all the years..

Do you know the term "scaffolding?"

What I did in 3.4 was put some scaffolding in while figuring out a big transformation.  At last the building is sound enough to take the scaffolding down again.

File: libraries/lib-wave-track/WaveTrack.cpp (hunk 1 above)
Comment (Paul-Licameli) [reply libraries/lib-wave-track/WaveTrack.cpp]: Wait to read a later commit that makes the distinction in return types.

File: src/menus/TrackMenus.cpp (hunk 2 above)
Comment (Paul-Licameli) [reply src/menus/TrackMenus.cpp]: Yes

In fact, I now think this commit needs to be in the previous PR for correctness of all stages of the transformation.

So I squashed it into a68333aad87a7dd417aa8ad73e5c8a8c3f8b5bbc

**Comment by Paul-Licameli** [comment]
Commit "Track::Duplicate and Clone return just a track, not a list" introduced a crashing bug, for instance when generating noise.  The fix was that `DuplicateWithOtherTempo` must remove the second, output argument, which was no longer getting initialized.  You may see this change in my next force-push of this branch.

**Review by saintmatthieu — APPROVED** [review]

**Comment by Paul-Licameli** [comment]
Rebased unchanged onto updated dependency

**Comment by Paul-Licameli** [comment]
Rebased unchanged onto updated wave-clip-refactoring branch

**Pull request merged**